dev/fattest.simplicity/src/componenttest/topology/utils/MvnUtils.java

```diff
@@ -70,296 +69,365 @@
+    }
 
-        if (!init) {
-            populateJarsFromWlp(pomXml);
-        }
+    /**
+     * runs "mvn clean test" in the tck folder, passing through all the required properties
+     *
+     * @param server the liberty server which should be used to run the TCK
+     * @param bucketName the name of the test project
+     * @param testName the name of the method that's being used to launch the TCK
+     * @param suiteFileName the name of the suite xml file
+     * @param additionalProps java properties to set when running the mvn command
+     * @param versionedJars A set of versioned jars
+     * @return the integer return code from the mvn command. Anything other than 0 should be regarded as a failure.
+     * @throws Exception occurs if anything goes wrong in setting up and running the mvn command.
+     */
+    public static int runTCKMvnCmd(LibertyServer server, String bucketName, String testName, String suiteFileName, Map<String, String> additionalProps,
+                                   Set<String> versionedJars) throws Exception {
+        MvnUtils mvn = new MvnUtils(server, bucketName, testName, suiteFileName, additionalProps, versionedJars);
+        return mvn.runCleanTest();
+    }
 
-        String mvn = "mvn";
-        if (System.getProperty("os.name").contains("Windows")) {
-            mvn = mvn + ".cmd";
-        }
-        mvnCliRaw = new String[] { mvn, "clean", "test", "-Dwlp=" + wlp, "-Dtck_server=" + server.getServerName(),
-                                   "-Dtck_failSafeUndeployment=" + DEFAULT_FAILSAFE_UNDEPLOYMENT,
-                                   "-Dtck_appDeployTimeout=" + DEFAULT_APP_DEPLOY_TIMEOUT,
-                                   "-Dtck_appUndeployTimeout=" + DEFAULT_APP_UNDEPLOY_TIMEOUT,
-                                   "-Dtck_port=" + server.getPort(PortType.WC_defaulthost),
-                                   "-DtargetDirectory=" + getResultsDir() + "/" + getTargetFolderName(),
-                                   "-DcomponentRootDir=" + componentRootDir,
-                                   "-Dsun.rmi.transport.tcp.responseTimeout=" + DEFAULT_MBEAN_TIMEOUT
-        };
-
-        mvnCliRoot = concatStringArray(mvnCliRaw, getJarCliProperties(server, jarsFromWlp));
+    /**
+     * Full constructor for MvnUtils. In most cases one of the static convenience methods should be used instead of calling this directly.
+     *
+     * @param server the liberty server which should be used to run the TCK
+     * @param bucketName the name of the test project
+     * @param testName the name of the method that's being used to launch the TCK
+     * @param suiteFileName the name of the suite xml file
+     * @param additionalProps java properties to set when running the mvn command
+     * @param versionedJars A set of versioned jars
+     */
+    public MvnUtils(LibertyServer server, String bucketName, String testName, String suiteFileName, Map<String, String> additionalMvnProps,
```

Comment:
Is this still a utils class or not?

It sort of is, because we seem to be only calling the static methods. If we _only_ want people to do that, should we make the constructor private?

Otherwise, if we expect other people to construct this object, should we change the name to reflect that it's no longer a utils class?

Reply:
well... it seems that in 70% of cases, the users of this class only ever called one method. So I provided some static convienence methods which still behave the same as they did before. We could split the static and non-static parts if you want ... not sure it is worth it??